whipper/common/program.py

```diff
@@ -614,3 +618,11 @@ def writeLog(self, discName, logger):
         self.logPath = logPath
 
         return logPath
+
+    def move_tocfile(self, discName):
+        """Move cdrdao's tocfile to the output dir. ('[discName].toc')"""
+        tocPath = '%s.toc' % discName
```

Comment:
~These variable names are confusing. Maybe something like `result_toc_path` to differentiate the new path from the temporary path.~

Honestly I think we might be doing this backwards. Maybe it's best to move the toc from /tmp to the result directory _in the `read_toc()` function itself_ if `persist_tocfile` is `True` and just `return toc` in `read_toc()` like before. I think it would keep things simpler.

Reply:
I agree.
Let's say I've done this:

```diff
diff --git a/whipper/program/cdrdao.py b/whipper/program/cdrdao.py
index e0da15c..778fb7a 100644
--- a/whipper/program/cdrdao.py
+++ b/whipper/program/cdrdao.py
@@ -1,5 +1,6 @@
 import os
 import re
+import shutil
 import tempfile
 from subprocess import Popen, PIPE
 
@@ -12,7 +13,7 @@ logger = logging.getLogger(__name__)
 CDRDAO = 'cdrdao'
 
 
-def read_toc(device, fast_toc=False):
+def read_toc(device, fast_toc=False, toc_path=None):
     """
     Return cdrdao-generated table of contents for 'device'.
     """
@@ -43,7 +44,10 @@ def read_toc(device, fast_toc=False):
 
     toc = TocFile(tocfile)
     toc.parse()
-    os.unlink(tocfile)
+    if toc_path is not None:
+        shutil.move(tocfile, toc_path)
+    else:
+        os.unlink(tocfile)
     return toc
 
 
@@ -83,14 +87,14 @@ def ReadTOCTask(device):
     """
     stopgap morituri-insanity compatibility layer
     """
-    return read_toc(device, fast_toc=True)
+    return read_toc(device, fast_toc=True, toc_path=None)
 
 
 def ReadTableTask(device):
     """
     stopgap morituri-insanity compatibility layer
     """
-    return read_toc(device)
+    return read_toc(device, toc_path=None)
 
 
 def getCDRDAOVersion():
```

Who gives me the proper `toc_path`? (unfortunately it seems that one isn't available both in `getFastToc()` and in `getTable()` methods).

P.S. I'm quite tired right now so I may be missing something obvious.

Reply:
From cd.py, pass the path to the function that ultimately calls read_toc(): something like `self.program.getTable(..., output_dir)` and `self.program.getFastToc(..., output_dir)`.

We probably don't want to save *both* the toc we've read with `getFastToc()` *and* `getTable()` though, just one?